src/components/TeleportPopover.vue

```diff
@@ -0,0 +1,382 @@
+<template>
+  <div>
+    <!-- Keep trigger in original location for layout -->
+    <div ref="triggerRef" class="teleport-popover-trigger">
+      <slot name="trigger" />
+    </div>
+    
+    <!-- Teleport custom popover content to preserve full Vue context -->
+    <Teleport to="#stac-browser">
+      <div
+        v-if="isVisible"
+        ref="popoverElement"
+        :class="['popover', 'bs-popover-' + placement, customClass]"
+        :style="popoverStyle"
+        role="tooltip"
+        @mouseenter="onPopoverMouseEnter"
+        @mouseleave="onPopoverMouseLeave"
+      >
+        <div class="popover-arrow" :style="arrowStyle" />
+        <h3 v-if="title" class="popover-header">{{ title }}</h3>
+        <div class="popover-body">
+          <slot name="content" />
+        </div>
+      </div>
+    </Teleport>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'TeleportPopover',
+  props: {
+    title: {
+      type: String,
+      default: ''
+    },
+    placement: {
+      type: String,
+      default: 'bottom',
+      validator: value => ['top', 'bottom', 'left', 'right'].includes(value)
```

Comment:
We never use validators yet, but it's a good idea. We should probably issue a separate PR that implements validators for all components where it's useful.